packages/shared/src/components/checklist/SharePostChecklistStep.tsx

```diff
@@ -1,17 +1,49 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useContext } from 'react';
 import { ChecklistStepProps } from '../../lib/checklist';
 import { Button } from '../buttons/Button';
 import { ChecklistStep } from './ChecklistStep';
 import { FlexRow } from '../utilities';
+import { useLazyModal } from '../../hooks/useLazyModal';
+import { LazyModal } from '../modals/common/types';
+import { Squad } from '../../graphql/sources';
+import OnboardingContext from '../../contexts/OnboardingContext';
 
-const SharePostChecklistStep = (props: ChecklistStepProps): ReactElement => {
-  // TODO WT-1293-checklist-components add business logic
+const SharePostChecklistStep = ({
+  squad,
+  ...props
+}: ChecklistStepProps & { squad: Squad }): ReactElement => {
+  const { openModal } = useLazyModal();
+  const { onInitializeOnboarding, showArticleOnboarding } =
+    useContext(OnboardingContext);
 
   return (
     <ChecklistStep {...props}>
       <FlexRow className="gap-4">
-        <Button className="btn-secondary">Create a post</Button>
-        <Button className="btn-primary">Explore</Button>
+        <Button
+          className="btn-secondary"
+          onClick={() => {
+            openModal({
+              type: LazyModal.PostToSquad,
+              props: {
+                ...props,
+                squad,
+                preview: { url: '' },
+              },
+            });
+          }}
+        >
+          Create a post
+        </Button>
+        {showArticleOnboarding && (
+          <Button
+            className="btn-primary"
+            onClick={() => {
+              onInitializeOnboarding();
+            }}
+          >
+            Explore
+          </Button>
+        )}
```

Comment:
If the user completed onboarding then we do not show explore button right? This could be a good check do differentiate new user maybe.

Do we even need separate action for this (we now have `ActionType.MyFeed`)? I know in some PR there were mentions of replacing `alerts` with this, but can not find it.

Reply:
Hmm, onboarding also finishes when user presses skip I think.
However we should have the action for this already as `my_feed` (Just not sure how it's set now, so maybe @sshanzel can mention this)

Reply:
`ActionType.MyFeed` gets set inside `updateAlerts` on the API side so it is covered, [reference change here](https://github.com/dailydotdev/daily-api/commit/6b1962a77337aefb0315355ebd2f52fbe34d6f83#diff-7632be2baf678486983af835e6be44d8f3d75a9641b49c9cc38942379b3a171eR114-R116)